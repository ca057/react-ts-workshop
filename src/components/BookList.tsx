import { Book } from "../domain/types";

import BookListItem from "./BookListItem";

interface BookListProps {
  books: Book[];
  onItemClick: (book: Book) => void;
}

const BookList: React.FC<BookListProps> = ({ books, onItemClick }) => {
  return (
    <ul>
      {books.map((book) => (
        <li key={book.title}>
          <BookListItem book={book} onClick={() => onItemClick(book)} />
        </li>
      ))}
    </ul>
  );
};

export default BookList;
