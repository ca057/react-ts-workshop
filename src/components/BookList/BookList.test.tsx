import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import BookList from ".";
import { BrowserRouter, MemoryRouter } from "react-router-dom";

describe("components/BookList", () => {
  // mock of a list of books
  const books = [
    { title: "My first book", price: 11.11 },
    { title: "My second book", price: 22.22 },
  ];

  test(`renders a booklist correctly`, () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <BookList books={books} />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // test case
  test("renders all books", () => {
    // render the booklist
    render(<BookList books={books} />, {
      // Define router as wrapper so that Jest can render the `Link` component from `react-router-dom`
      wrapper: BrowserRouter,
    });

    for (const book of books) {
      // assert that each book will be rendered to the screen
      expect(screen.getByText(book.title)).toBeTruthy();
    }
  });
});
