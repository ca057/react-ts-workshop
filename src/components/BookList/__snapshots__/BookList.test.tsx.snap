// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/BookList renders a booklist correctly 1`] = `
<ul>
  <li>
    <a
      href="/books/undefined"
      onClick={[Function]}
    >
      <div
        className="booklist__item"
      >
        My first book
      </div>
    </a>
  </li>
  <li>
    <a
      href="/books/undefined"
      onClick={[Function]}
    >
      <div
        className="booklist__item"
      >
        My second book
      </div>
    </a>
  </li>
</ul>
`;
